### PR TITLE
client_icons.txt update

### PR DESCRIPTION
**Pull request by Neustradamus**
client_icons.txt update

### CODE REVIEW

**Comment by Neustradamus** [comment]
Have you looked?
Already more 10 days.

**Comment by tehnick** [comment]
Have you checked each client from this list which caps you have modified?

**Comment by Neustradamus** [comment]
Any news for this PR?
Of course it is good.

**Comment by Ri0n** [comment]
after some discussion it turned out the changes were done assuming any part of caps can be used for match. 
That's wrong. THe right part of a fingerprint spec must always start from the beginning of caps of client name. And if something in the middle is required for clarification then it has to be prefixed with "#" symbol. And if "#" will be used everywhere, this make the lookup table pretty much not effective downgrading lookups from O(log n) to O(N).

So the proper change should follow next rules (in the order of preference)

1. Use client name  (or its beginning part) for mature clients if a client supports caps and diso#info returns client name in "urn:xmpp:dataforms:softwareinfo" xmpp form.
2. Use caps node as is or it's beginning part if client name is not provided in disco#info or client name is a matter to change.
3. Use client name if no caps provided and the cliet name is returned either with disco or version query.

If a caps node for old clients starts with http and for new clients with https, all of them have to be presented.

Example for caps-only client
```
clientx: http://clientx.com,https://clientx.com
```

**Comment by Ri0n** [comment]
> rion:
> you literally installed them and checked detection?
> ￼
> Neustradamus:
> Yes
> ￼
> rion:
> wow. seems like you have really a lot of spare time

I guess we can merge it then.

**Comment by Ri0n** [comment]
I was a little bit wrong in my previous comment. There just one icon lookup by client name. But if client name is unknown it will treat caps node as an url and will take its host part.
It's the current implementation in UserListItem::findClient. not perfect but that's it.

File: client_icons.txt [hunk 1]
```diff
@@ -111,53 +97,53 @@ oneteamiphone process-one.net
...
-qipmobile mobileqip.ru &&qip 
-qipinfium qip#infium,qip#2010,qip#2012,2010.qip.ru
+pyicq-t pyicq,icq#transport
+qip qip
+qq qq
 qutim qutim
-radio-t apps.radio-t.com
-riddim code.matthewwild.co.uk#riddim
+qxmpp qxmpp
+radio-t radio-t
```
Comment (Ri0n) [inline client_icons.txt]: as far as I understand it's not supported anymore

File: client_icons.txt [hunk 2]
```diff
@@ -111,53 +97,53 @@ oneteamiphone process-one.net
...
-utalk palringo.com
+uwpx uwpx
+vacuum vacuum
+vk4xmpp vk4xmpp,simpleapps.ru
+vkontakte vkontakte,vk.com,pyvk
 weather.com jabber weather.com#transport
-webclient chat.jabbercity.ru,web-am31.dyndns-ip.com
-weonlydo weonlydo
-wod-xmpp weonlydo.com
+wodxmpp weonlydo
```
Comment (Ri0n) [inline client_icons.txt]: why have you changed icon name?

File: client_icons.txt [hunk 3]
```diff
@@ -111,53 +97,53 @@ oneteamiphone process-one.net
...
 wtw wtw
-vacuum vacuum
-vkontakte vk.com,pyvk-t,vkontakte
 yabber yabber
+yandex yandex
```
Comment (Ri0n) [inline client_icons.txt]: it doen't exist anymore

File: client_icons.txt (hunk 2 above)
Comment (Neustradamus) [reply client_icons.txt]: -> https://www.weonlydo.com/XMPP/xmpp-icq-msn-component.asp

File: client_icons.txt (hunk 3 above)
Comment (Neustradamus) [reply client_icons.txt]: -> http://jabberworld.info/Jabber_caps

Comment (Ri0n) [reply client_icons.txt]: Ya.Online is dead for years. This entry really has to be removed

File: client_icons.txt (hunk 2 above)
Comment (Ri0n) [reply client_icons.txt]: I don't care about the code. We have wod-xmpp icon, not wodxmpp

Comment (Ri0n) [reply client_icons.txt]: well real name doesn't matter too until you decide to update iconpacks too.

File: client_icons.txt (hunk 1 above)
Comment (Ri0n) [reply client_icons.txt]: drop this record

File: client_icons.txt (hunk 3 above)
Comment (msva) [reply client_icons.txt]: Btw, there is YaPlus, and it looks like there is some kind of jabber there, but I didn't tested

**Comment by Neustradamus** [comment]
@Ri0n Can you look this time?

**Comment by Ri0n** [comment]
@Neustradamus this PR is way to huge and has a lot of suspicious changes. 
I propose to split it to smaller ones.
Like 
1) removing died clients
2) change detecting of existing
3) add new clients

When you change detecting it's better to refer to the source of "why"

After this split it will have way more chances to merge.

**Pull request merged**